application/src/main/scala/com/azavea/franklin/commands/ApiOptions.scala

```diff
@@ -8,52 +8,83 @@ import eu.timepit.refined.types.numeric.PosInt
 
 trait ApiOptions {
 
+  private val portDefault      = PosInt(9090)
+  private val externalPortHelp = s"Port users/clients hit for requests. Default: '$portDefault'."
+
   val externalPort = Opts
-    .option[PosInt]("external-port", help = "Port users/clients hit for requests")
-    .withDefault(PosInt(9090))
+    .option[PosInt]("external-port", help = externalPortHelp) orElse
+    Opts.env[PosInt]("API_EXTERNAL_PORT", help = externalPortHelp) withDefault (portDefault)
+
+  private val internalPortHelp =
+    s"Port server listens on, this will be different from 'external-port' when service is started behind a proxy. Default: '$portDefault'."
 
   val internalPort = Opts
     .option[PosInt](
       "internal-port",
-      help =
-        "Port server listens on, this will be different from 'external-port' when service is started behind a proxy"
-    )
-    .withDefault(PosInt(9090))
+      help = internalPortHelp
+    ) orElse Opts
+    .env[PosInt]("API_INTERNAL_PORT", help = internalPortHelp) withDefault (portDefault)
+
+  private val apiHostDefault = "localhost"
+
+  private val apiHostHelp =
+    s"Hostname Franklin is hosted it (e.g. localhost). Default: '$apiHostDefault'."
 
   val apiHost = Opts
-    .option[String]("api-host", help = "Hostname Franklin is hosted it (e.g. localhost)")
-    .withDefault("localhost")
+    .option[String]("api-host", help = apiHostHelp) orElse
+    Opts.env[String]("API_HOST", help = apiHostHelp) withDefault (apiHostDefault)
+
+  private val apiPathHelp = "Path component for root of Franklin instance (e.g. /stac/api)."
 
   val apiPath = Opts
     .option[String](
       "api-path",
-      help = "Path component for root of Franklin instance (e.g. /stac/api)"
-    )
-    .orNone
+      help = apiPathHelp
+    ) orElse
+    Opts.env[String]("API_PATH", help = apiPathHelp) orNone
+
+  private val apiSchemeDefault = "http"
+
+  private val apiSchemeHelp =
+    s"Scheme server is exposed to end users with. Default: '$apiSchemeDefault'."
 
   val apiScheme =
     Opts
-      .option[String]("api-scheme", "Scheme server is exposed to end users with")
-      .withDefault("http")
-      .validate("Scheme must be either 'http' or 'https'")(s => (s == "http" || s == "https"))
+      .option[String]("api-scheme", help = apiSchemeHelp) orElse
+      Opts.env[String]("API_SCHEME", help = apiSchemeHelp) withDefault ("http")
+  validate("Scheme must be either 'http' or 'https'")(s => (s == "http" || s == "https"))
```

Comment:
Also, based on the CI error, I think [you'll need to provide](http://ben.kirw.in/decline/arguments.html#missing-instances) a custom `Argument` instance for `Boolean` -- but it might make more sense just to leave the flags CLI only. I have a pretty ad hoc distinction in my head between environment variables, which describe the state of the system Franklin's running in, and CLI flags, which tell Franklin what it should do, and that ad hoc distinction says to let the flags be CLI flags only.

Reply:
Thanks, I made sure to combine these two Opts in updated code